jraft-core/src/main/java/com/alipay/sofa/jraft/core/ReadOnlyServiceImpl.java

```diff
@@ -116,16 +123,23 @@ public void onEvent(final ReadIndexEvent newEvent, final long sequence, final bo
                                                                                                          throws Exception {
             if (newEvent.shutdownLatch != null) {
                 executeReadIndexEvents(this.events);
-                this.events.clear();
+                reset();
                 newEvent.shutdownLatch.countDown();
                 return;
             }
 
             this.events.add(newEvent);
             if (this.events.size() >= ReadOnlyServiceImpl.this.raftOptions.getApplyBatch() || endOfBatch) {
                 executeReadIndexEvents(this.events);
-                this.events.clear();
+                reset();
+            }
+        }
+
+        private void reset() {
+            for (ReadIndexEvent event : this.events) {
```

Comment:
for (final ReadIndexEvent event : this.events) {

Reply:
done